src/CodeGen_Hexagon.cpp

```diff
@@ -1297,8 +1312,17 @@ void CodeGen_Hexagon::visit(const Div *op) {
 }
 
 void CodeGen_Hexagon::visit(const Cast *op) {
-    // TODO: Do we need to handle same-sized vector casts before LLVM sees them?
-    CodeGen_Posix::visit(op);
+    if (op->value.type().bits() == 1 && op->type.bits() > 1) {
+        // If left to LLVM, this maps true to one. However,
+        // eliminate_bool_vectors assumes that it maps to negative
+        // one. This will map true to negative one instead.
+        value = call_intrin(op->type,
```

Comment:
I realized another thing too: the comment is wrong, it doesn't currently produce one for true; it currently returns a value where each byte of the result is one. I don't think there's a way to make booleans 0 or 1 efficiently, because vmux is not typed.